runtime/common/RuntimeMLIR.cpp

```diff
@@ -258,6 +258,67 @@ mlir::LogicalResult verifyOutputRecordingFunctions(llvm::Module *llvmModule) {
   return success();
 }
 
+// Convert a `nullptr` or `inttoptr (i64 1 to Ptr)` into an integer
+std::size_t getArgAsInteger(llvm::Value *arg) {
+  std::size_t ret = 0; // handles the nullptr case
+  // Now handle the `inttoptr (i64 1 to Ptr)` case
+  auto constValue = dyn_cast<llvm::Constant>(arg);
+  if (auto constExpr = dyn_cast<llvm::ConstantExpr>(constValue))
+    if (constExpr->getOpcode() == llvm::Instruction::IntToPtr)
+      if (auto constInt = dyn_cast<llvm::ConstantInt>(constExpr->getOperand(0)))
+        ret = constInt->getZExtValue();
```

Comment:
Does QIR forbid using signed values?

Reply:
> Does QIR forbid using signed values?

Yes. The spec says: The integer constant that is cast must be in the interval `[0, numQubits)` for `%Qubit*` and `[0, numResults)` for `%Result*`, where `numQubits` and `numResults` are the required number of qubits and results defined by the corresponding entry point attributes.